modules/virt-accessing-vmi-yaml-ssh.adoc

```diff
@@ -0,0 +1,130 @@
+
+The following examples show the configurations for the VMI's YAML file and the service YAML file.
+
+.Prerequisites
+* Install the OpenShift CLI (`oc`).
+* You must be in the same project as the VMI.
+* The VMI you want to access must be connected to the default pod network by using the `masquerade` binding method.
+* The VMI you want to access must be running.
+
+.Procedure
+. Edit the YAML file for the VMI. Specify a name for exposing the service for SSH connections and the port number. Enable the `masquerade` feature for the interface.
+
++
+.Example output
+[source,yaml]
+----
+...
+kind: VirtualMachineInstance
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vmi-ssh
+  name: vmi-ssh <1>
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      interfaces:
+      - masquerade: {}
+        name: testmasquerade <2>
+        ports:
+        - name: ssh
+          port: 22 <3>
+          protocol: TCP
+      rng: {}
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 1024M
+  networks:
+  - name: testmasquerade
+    pod: {}
+...
+----
+<1> The name for the VMI that is used to expose the details for the SSH service.
+<2> The name of the `masquerade` for the interface.
+<3> The port number used for SSH connections.
+
+. Verify that the SSH service is running:
++
+[source,terminal]
+----
+oc get vmi
+----
++
+.Example output
+[source,terminal]
+----
+NAME    AGE     PHASE       IP              NODENAME
+vmi-ssh 6s      Running     10.244.196.152  node01
+----
+
+. Check the service to find out which port was acquired by the service:
++
+[source,terminal]
+----
+$ oc get svc
+----
++
+.Example output
+[source,terminal]
+----
+NAME            TYPE       CLUSTER-IP     EXTERNAL-IP   PORT(S)           AGE
+svc-ssh-exp     NodePort   10.106.236.208 <none>        20022:30093/TCP   22s
+----
++
+In this example, the service acquired the port number 20022.
+
+. Create the YAML file for the SSH service:
++
+[source,terminal]
+----
+$ oc create -f vmi-ssh.yaml
+----
+
++
+.Example output
+[source,yaml]
+----
+...
+apiVersion: v1
+kind: Service
+metadata:
+  name: svc-ssh <1>
+spec:
+  ports:
+    - port: 20022 <2>
+      targetPort: 22
+      protocol: TCP
+  selector:
+    special: vmi-ssh <3>
+  type: NodePort
+...
+----
+<1> Name of the service that was specified in the YAML file for the VMI.
+<2> The port number that is assigned to direct SSH connections to the VMI.
+<3> Name for the VMI that exposes the service for SSH connections.
+
+. Log in to the VMI via SSH. Use the `ipAddress` of any of the cluster nodes and the port that you found in the previous step:
++
+[source,terminal]
+----
+$ ssh username@<node_IP_address> -p port_number
```

Comment:
maybe change port_number to an explicit example like: 
30093
?

Reply:
Here, we are instructing the user to insert their IP address and port number. I've added brackets around the port number to indicate that the user enters their own port number. Let me know if that works!

Reply:
I need to troubleshoot why <port_number> is not displaying in the HTML content. Looking into this.